app/models/miq_report.rb

```diff
@@ -130,8 +130,7 @@ def self.display_filter_details(cols, mode)
     return [] if cols.nil?
     cols.inject([]) do |r, c|
       # eg. c = ["Host.Hardware.Disks : File Name", "Host.hardware.disks-filename"]
-      parts = c.last.split(".")
-      parts[-1] = parts.last.split("-").first # Strip off field name from last element
+      *parts, _ = MiqExpression::Field.parse(c.last).to_a
```

Comment:
This seems a little odd. Having done some parsing in `Field`, we smash the parts back together and then do some more parsing below. Is there any way that the code below could leverage more of `Field`, or enhance it, so we don't need a `to_a` method?